content/en/docs/File Format/implementationstatus.md

```diff
@@ -45,64 +45,63 @@ Implementations:
-| 8, 16, 32, 64 bit signed and unsigned INT |       |       |       |       |  ✅   |
-| DECIMAL (INT32)                           |       |       |       |       |  ✅   |
-| DECIMAL (INT64)                           |       |       |       |       |  ✅   |
-| DECIMAL (BYTE_ARRAY)                      |       |       |       |       |  ✅   |
-| DECIMAL (FIXED_LEN_BYTE_ARRAY)            |       |       |       |       |  ✅   |
-| DATE                                      |       |       |       |       |  ✅   |
-| TIME (INT32)                              |       |       |       |       |  ✅   |
-| TIME (INT64)                              |       |       |       |       |  ✅   |
-| TIMESTAMP (INT64)                         |       |       |       |       |  ✅   |
-| INTERVAL                                  |       |       |       |       |  ❌   |
-| JSON                                      |       |       |       |       |  ❌   |
-| BSON                                      |       |       |       |       |  ❌   |
-| LIST                                      |       |       |       |       |  ✅   |
-| MAP                                       |       |       |       |       |  ✅   |
-| UNKNOWN (always null)                     |       |       |       |       |  ✅   |
-| FLOAT16                                   |       |       |       |       |  ✅   |
+| STRING                                    |  ✅   |  ✅   |       |       |  ✅   |
+| ENUM                                      |  ❌   |  ✅   |       |       |  ❌   |
+| UUID                                      |  ❌   |  ✅   |       |       |  ❌   |
+| 8, 16, 32, 64 bit signed and unsigned INT |  ✅   |  ✅   |       |       |  ✅   |
+| DECIMAL (INT32)                           |  ✅   |  ✅   |       |       |  ✅   |
+| DECIMAL (INT64)                           |  ✅   |  ✅   |       |       |  ✅   |
+| DECIMAL (BYTE_ARRAY)                      |  ✅   |  ✅   |       |       |  ✅   |
+| DECIMAL (FIXED_LEN_BYTE_ARRAY)            |  ✅   |  ✅   |       |       |  ✅   |
+| DATE                                      |  ✅   |  ✅   |       |       |  ✅   |
+| TIME (INT32)                              |  ✅   |  ✅   |       |       |  ✅   |
+| TIME (INT64)                              |  ✅   |  ✅   |       |       |  ✅   |
+| TIMESTAMP (INT64)                         |  ✅   |  ✅   |       |       |  ✅   |
+| INTERVAL                                  |  ✅   |  ✅   |       |       |  ❌   |
+| JSON                                      |  ✅   |  ✅   |       |       |  ❌   |
+| BSON                                      |  ❌   |  ✅   |       |       |  ❌   |
+| LIST                                      |  ✅   |  ✅   |       |       |  ✅   |
+| MAP                                       |  ✅   |  ✅   |       |       |  ✅   |
+| UNKNOWN (always null)                     |  ✅   |  ✅   |       |       |  ✅   |
+| FLOAT16                                   |  ✅   |  ✅   |       |       |  ✅   |
```

Comment:
What do we mean by logical types are supported? Is it like we won't throw an exception and return something at least, or we expect to actually represent the related value correctly?
Since `parquet-java` does not have its own representation type system (except the one in the example package that mainly handles the primitive types only), the question is whether at least one binding supports the related type. I am not aware that any binding supports `INTERVAL` or `BSON`. `JSON` is mostly handle as string without any syntax check.

Reply:
I don't think won't throw should be considered as supported. Let me remove `INTERVAL` and `BSON`.

Just to confirm that `ENUM` and `UUID` are supported by the Avro binding, right?

Reply:
Yes, `UUID` is mapped to the Avro UUID type, `ENUM` is mapped to string.

But, `JSON` and `FLOAT16` do not seem to be supported either.

Reply:
I think `FLOAT16` should be considered as supported: https://github.com/apache/parquet-java/pull/1142

Reply:
So, that's why I asked what supporting a logical type mean in terms of `parquet-java` where we do not have our own representation type system. For `FLOAT16`, we return the containing 2 bytes as a `Binary` in our example package. We do not map this type in any of our bindings.
If we consider `FLOAT16` as supported, then all the others are supported as well.

Reply:
Is there value in adding another option here (`P` perhaps) that means the implementation at least supports the logical type enum value but will return the physical type? As we add logical types to Parquet I think it's useful to know which implementations will at least not error out on more recent additions to the spec.

Reply:
> If we consider FLOAT16 as supported, then all the others are supported as well.

Yes, that's why I marked all logical types as supported in the initial commit. Developers are able to implement their own bindings to write and read such types by extending the `parquet-java` library.

> Is there value in adding another option here (P perhaps) that means the implementation at least supports the logical type enum value but will return the physical type?

I'm not sure. Maybe it is a minimum requirement for implementation?

Reply:
@wgtmac, If we do not want to distinguish the support level of e.g. `FLOAT16` and `UUID`, I'm fine having all marked as supported.

Reply:
OK, fixed.

Reply:
> Yes, that's why I marked all logical types as supported in the initial commit. Developers are able to implement their own bindings to write and read such types by extending the `parquet-java` library.

You should then add a note or asterisk to point out that "supported" means the physical type is returned.

Reply:
Sounds good. Added asterisk to explain that.